gsoc/admin.py

```diff
@@ -396,13 +397,23 @@ def email(self, obj):
         return obj.user.email
 
     def get_queryset(self, request):
-        return UserProfile.all_objects.all()
+        """
+        This queryset is used to return UserProfile object based on the role of request.user
+        If the user is a super user then all users are returned.
+        If the user is a suborg admin then mentors and students of that suborg are returned.
+        """
+        user = request.user.userprofile_set.get()
+        if request.user.is_superuser:
+            return UserProfile.all_objects.all()
+        if user.role == 1:
+            return UserProfile.objects.filter(suborg_full_name=user.suborg_full_name)
```

Comment:
you could set this filter to be the current year, as well as the suborg in question at the same time, doing so would make the filters.py redundant

Reply:
For suborg admins only current year mentors are shown for psf admins all the users are shown sorted by year.